agent/src/util/config_action.rs

```diff
@@ -276,6 +291,24 @@ async fn handle_config_delete(
     Ok(())
 }
 
+/// Checks to see if the configuration needs to be recreated.
+/// At present, this just checks to see if the `.metadata.generation` has changed.
+/// The `.metadata.generation` value is incremented for all changes, except for changes to `.metadata` or `.status`.
+async fn should_recreate_config(config: &Configuration, config_map: ConfigMap) -> Result<bool, ()> {
+    let name = config.metadata.name.clone().unwrap();
+    let last_generation = config_map.lock().await.get(&name).unwrap().last_generation;
```

Comment:
There is probably some sort of error handling that should happen around this where we get the config from the map. Still trying to wrap my head around how that works in Rust.

Reply:
It would be best to bubble up this error instead of potentially panicking with an `unwrap`. Maybe return type `Result<bool, anyhow::Error>`  and do something like 
`let last_generation = config_map.lock().await.get(&name).ok_or(anyhow::anyhow!("Configuration {} not found in ConfigMap", name))?.last_generation;`

Reply:
Looks like clippy doesn't love the ok_or usage...
```
warning: use of `ok_or` followed by a function call
   --> agent/src/util/config_action.rs:304:10
    |
304 |           .ok_or(anyhow::anyhow!(
    |  __________^
305 | |             "Configuration {} not found in ConfigMap",
306 | |             name
307 | |         ))?
    | |__________^
    |
    = note: `#[warn(clippy::or_fun_call)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call
help: try this
    |
304 |         .ok_or_else(|| anyhow::anyhow!(
305 |             "Configuration {} not found in ConfigMap",
306 |             name
307 |         ))?
    |
```

Gonna try its suggestion.

Reply:
Looks like it seems happier about that.